### PR TITLE
4.0.0-rc2

### PR DESCRIPTION
**Pull request by zTrap**
- Library fully rewritten in _Kotlin_
- typeface libraries now have a new package name pattern: `com.mikepenz.iconics.typeface.library.<libraryname>.<TypefaceClass>`
- updated proguard rules
- introduced context resolving via the application `Context` using a `ContentProvider`
- removed the required provided `Context` in some methods / classes
- added migration information
 - made all typeface libraries are `kotlin objects`
 - added extensions to adapt `Iconics.Builder`
 - removed old `iconicsBuilder` extensions
 - added providable logger interface
 - added resource which enable / disable `IconicsContentProvider` (see `R.bool.is_iconics_content_provider_enabled`)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -83,7 +83,7 @@ implementation 'com.mikepenz:material-design-iconic-typeface:2.2.0.6-kotlin@aar'
 implementation 'com.mikepenz:fontawesome-typeface:5.3.1.2-kotlin@aar'
 implementation 'com.mikepenz:octicons-typeface:3.2.0.6-kotlin@aar'
 implementation 'com.mikepenz:meteocons-typeface:1.1.0.5@aar'
-implementation 'com.mikepenz:community-material-typeface:3.5.95.2-kotlin@aar'
```
Comment (mikepenz) [inline README.md]: @zTrap I think you reverted that by accident

File: app/src/main/java/com/mikepenz/iconics/sample/IconsFragment.kt [hunk 2]
```diff
@@ -54,69 +57,65 @@ import kotlin.math.abs
...
         // Init and Setup RecyclerView
-        val recyclerView = list
-        recyclerView.layoutManager = GridLayoutManager(activity, 2)
-        recyclerView.addItemDecoration(SpaceItemDecoration())
-        //animator not yet working
-        recyclerView.itemAnimator = DefaultItemAnimator()
-        mAdapter = FastItemAdapter()
-        configAdapter()
-        recyclerView.adapter = mAdapter
+        list.also {
```
Comment (mikepenz) [inline app/src/main/java/com/mikepenz/iconics/sample/IconsFragment.kt]: we can also use `apply` here to get rid of the `it`

File: README.md (hunk 1 above)
Comment (zTrap) [reply README.md]: Currently we have 
`implementation 'com.mikepenz:community-material-typeface:2.7.94.2-kotlin@aar'` and
`implementation 'com.mikepenz:community-material-typeface:3.5.95.1@aar'` versions. I've choose to greater version

Comment (zTrap) [reply README.md]: I think updating this part of README is part of flow of releasing new version :)

Comment (mikepenz) [reply README.md]: Oh I see. :D yeah works, just remember that the 3.5.x is not compatible with v4 only v2.7.94-2 as it was released for kotlin :D

**Comment by rubengees** [comment]
Just tried the latest version and the `IconicsMenuInflaterUtil` does not work correctly anymore. The icons are completely transparent (but can still be clicked).

My `menu`:

```xml
<item
    android:id="@+id/filter"
    android:title="@string/action_filter"
    app:ico_color="@android:color/white"
    app:ico_icon="cmd_filter_variant"
    app:ico_padding="2dp"
    app:ico_size="24dp"
    app:showAsAction="ifRoom" />
```

I also tried setting a color directly which has no effect: `app:ico_color="#FFFFFF"`
Setting other colors (e.g. `app:ico_background_color="#FFFFFF"`) works.

**Pull request closed without merging**